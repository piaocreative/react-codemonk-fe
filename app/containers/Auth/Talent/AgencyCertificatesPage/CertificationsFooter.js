import React from 'react';
import { FormattedMessage } from 'react-intl';
import { LinkButtonMod, Button } from 'components';
import { handleBackButton } from 'containers/Auth/utils';
import { propTypes } from 'containers/proptypes';
import messages from './messages';

const CertificationsFooter = ({ history, invalid, loading, handleSubmit, handleSaveForLater }) => (
  <div className="d-flex justify-content-md-end align-items-center flex-wrap justify-content-between">
    <LinkButtonMod
      className="left-arrow link me-auto"
      color="link"
      onClick={e => {
        handleBackButton(e, history, '/agency/add-talents');
      }}
    >
      <FormattedMessage {...messages.btnBack} />
    </LinkButtonMod>

    <LinkButtonMod
      color="link"
      onClick={e => {
        handleSaveForLater(e, 'saveForLater');
      }}
    >
      <FormattedMessage {...messages.btnSaveLater} />
    </LinkButtonMod>
    <Button
      className={`${loading ? 'loading' : ''} btn-primary btn-submit`}
      disabled={invalid}
      onClick={handleSubmit(e => handleSaveForLater(e, 'continue'))}
    >
      {messages.continueBtn.defaultMessage}
    </Button>
  </div>
);
export default CertificationsFooter;

CertificationsFooter.propTypes = propTypes;
