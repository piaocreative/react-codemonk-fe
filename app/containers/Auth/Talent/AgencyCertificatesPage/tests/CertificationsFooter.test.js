import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CertificationsFooter from '../CertificationsFooter';

const props = {
  certificateDetails: [
    {
      name: 'AWS Solution Architect',
      dateObtained: '30/08/2019',
      issuedBy: 'Amazon',
    },
  ],
  dispatch: jest.fn(),
  handleSubmit: jest.fn(),

  invalid: false,
  loading: false,
  responseSuccess: false,
  responseError: false,
  onReset: jest.fn(),
  history: { push: jest.fn(), replace: jest.fn() },

  onChangeCertificateDetails: jest.fn(),
  onSaveForLater: jest.fn(),
  onSubmitEducationForm: jest.fn(),

  onAddCertificateForm: jest.fn(),
  onDeleteCertificateForm: jest.fn(),
};

describe('<CertificationsFooter />', () => {
  const renderer = new ShallowRenderer();
  it('should render and match the snapshot', () => {
    renderer.render(<CertificationsFooter {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
