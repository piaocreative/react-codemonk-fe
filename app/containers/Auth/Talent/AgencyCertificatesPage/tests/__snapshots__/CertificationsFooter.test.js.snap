// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CertificationsFooter /> should render and match the snapshot 1`] = `
<div
  className="d-flex justify-content-md-end align-items-center flex-wrap justify-content-between"
>
  <ForwardRef(LinkButtonMod__ButtonMod)
    className="left-arrow link me-auto"
    color="link"
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="Back"
      id="CodeMonk.components.agencyCertificateForm.btnBack"
      values={Object {}}
    />
  </ForwardRef(LinkButtonMod__ButtonMod)>
  <ForwardRef(LinkButtonMod__ButtonMod)
    color="link"
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="Save for Later"
      id="CodeMonk.components.agencyCertificateForm.btnSaveLater"
      values={Object {}}
    />
  </ForwardRef(LinkButtonMod__ButtonMod)>
  <Button
    className=" btn-primary btn-submit"
    disabled={false}
  >
    Continue
  </Button>
</div>
`;
